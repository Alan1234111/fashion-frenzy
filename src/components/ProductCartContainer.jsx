import Button from "react-bootstrap/esm/Button";

function ProductCartContainer(props) {
  console.log(props.price);
  return (
    <div className="d-flex align-items-center justify-content-between p-4 border-top border-bottom" style={{height: "150px"}}>
      <img style={({height: "100px"}, {width: "10%"})} src={props.img} alt="" />
      <p className="fw-bold m-0 w-25">{props.title}</p>
      <div className="d-flex align-items-center justify-content-center">
        <Button className="text-secondary bg-white border-0 fs-4">-</Button>
        <input type="text" className="item-quantity-input text-center" value={props.quantity} />
        <Button className="text-secondary bg-white border-0">+</Button>
      </div>
      <p className="fw-bold m-0" style={{width: "70px"}}>
        $ {props.price.toFixed(2)}
      </p>
      <Button onClick={() => props.handleDelete(props.id)} className="text-secondary bg-white border-0">
        &#10005;
      </Button>
    </div>
  );
}

export default ProductCartContainer;
