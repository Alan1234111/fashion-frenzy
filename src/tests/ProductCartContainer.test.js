import React from "react";
import {render} from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";
import ProductCartContainer from "../components/ProductCartContainer";

test("renders ProductCartContainer correctly", () => {
  const props = {
    id: "123",
    img: "image-url",
    title: "Product Title",
    quantity: 2,
    price: 9.99,
    decrement: jest.fn(),
    increment: jest.fn(),
    handleDelete: jest.fn(),
  };
  const {container} = render(
    <Router>
      <ProductCartContainer {...props} />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
