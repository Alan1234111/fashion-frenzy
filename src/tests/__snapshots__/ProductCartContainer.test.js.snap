// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductCartContainer correctly 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-between p-4 border-top border-bottom flex-column flex-md-row container"
  >
    <div
      class="p-4 p-md-0 col-md-2"
    >
      <a
        href="/products/123"
      >
        <img
          alt=""
          src="image-url"
          style="height: 100px;"
        />
      </a>
    </div>
    <div
      class="col-md-3"
    >
      <p
        class="fw-bold m-0"
      >
        Product Title
      </p>
    </div>
    <div
      class="d-flex align-items-center justify-content-center col-md-4"
    >
      <button
        class="text-secondary bg-white border-0 fs-4 btn btn-primary"
        type="button"
      >
        -
      </button>
      <input
        class="item-quantity-input text-center"
        type="number"
        value="2"
      />
      <button
        class="text-secondary bg-white border-0 btn btn-primary"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="col-md-2"
    >
      <p
        class="fw-bold m-md-0 mt-2"
        style="width: 70px;"
      >
        $ 
        9.99
      </p>
    </div>
    <div
      class="col-md-2"
    >
      <button
        class="text-secondary bg-white border-0 btn btn-primary"
        type="button"
      >
        âœ•
      </button>
    </div>
  </div>
</div>
`;
